### PR TITLE
Add watch support to makefile

### LINKED ISSUE #375: Look into updating build to support watch
Currently we need to do a full make clean install to get a consistent experience during development. We want to look into what it would take to update our makefile to instead support `jupyter lab --watch` which would allow us to edit our front end packages and see the changes without a full rebuild.

### PR DESCRIPTION
**Pull request by ajbozarth**
Fixes #375
Partially addresses #501

Includes:
- Updates makefile to link packages instead of creating dists and
installing them (allows for use of `watch`)
- Add make  watch command that watches for changes and rebuilds
packages. To be used alongside `jupyter lab --watch`

Known issues:
- `jupyter lab --watch` cant pick up changes in dependencies within
a monorepo, meaning it will not pick up changes in `application`
when called within another extension. A simple workaround is to
stop and start lab, changes in application do get picked up on start

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by ajbozarth** [comment]
@akchinSTC and @lresende I am unsure how removing the creation of the dists will affect other parts of the code. I was unable to figure out if those dists were leveraged anywhere other than the `install` step of the Makefile. If they are needed elsewhere I can add a new make target that builds them outside the make install flow.

**Comment by ajbozarth** [comment]
With this PR you can utilize watch with the following steps:
1. Initial build with `make clean install`
2. Watch for changes with `make watch`
3. Run Lab with `jupyter lab --watch`

The caveats:
- Steps 2 and 3 above tail and need their own terminal tab/window (make sure the conda env is the same)
- Changes to utils in application that are called within other packages will not be picked up by `jupyter lab --watch` and require stopping and starting to pick up.
- When running `jupyter lab --watch` the start up of lab takes about 2-3x longer.

**Comment by ajbozarth** [comment]
After some clarification from @akchinSTC offline I've added back in the functionality to package the extensions into dists since they are included when creating elyra releases. The added make target assumes `make install` has already been run.

On a separate note, I updated the `.PHONY` targets in this PR. new targets were added to Makefile but were missing from `.PHONY` also since it was getting long I split it across two lines.

**Comment by ajbozarth** [comment]
Per discussion in scrum I will be adding documentation for this. @lresende where should that documentation go? In the README, in the new docs we just added, or both?

**Comment by lresende** [comment]
The documentation should be added to where we describe the contributor workflow, currently, it's available in [README.md](https://github.com/elyra-ai/elyra#development-workflow) and on the [development workflow](https://github.com/elyra-ai/elyra/blob/master/docs/source/developer_guide/development-workflow.md)

**Comment by kevin-bates** [comment]
Could we please have a single location for dev workflow and link README to the docs?  

Then promote the community meeting section header up a level and add a sentence indicating/inviting folks to attend if they're interested in contributing?

Let's also remove the single ticks around `Elyra` in the opening dev workflow paragraph since this is more about the project and not an object within the project.

If anyone feels this needs to be its own separate pull request, then I'm happy to submit a PR later today.

**Comment by ajbozarth** [comment]
@kevin-bates if you want to open another pr for that then I will only update the docs in this pr

**Comment by kevin-bates** [comment]
Thanks Alex - I'll a PR with the changes I proposed above.

**Comment by ajbozarth** [comment]
@lresende should we address the handling of dists in #503 rather than handling it here? ie once I push the docs update this would be ready to merge?

**Comment by ajbozarth** [comment]
I've pushed docs for review

**Comment by marthacryan** [comment]
Just played around with this locally and it worked for me!

**Comment by karlaspuldaro** [comment]
Worked for me too! This is great, gonna help a lot our dev process :)

File: docs/source/developer_guide/development-workflow.md [hunk 1]
```diff
@@ -42,3 +42,25 @@ You can check that the JupyterLab extension was successful installed with:
...
+
+Elyra supports incremental development using `--watch`. This allows you to make code changes to
+front-end packages and see them without running `make install` again.
+
+After installation run the following to watch for code changes and rebuild automatically:
+```bash
+make watch
+```
+
+Then in a separate terminal using the same environment, start JupyterLab in watch mode:
```
Comment (vabarbosa) [inline docs/source/developer_guide/development-workflow.md:56]: is it worth clarifying what you mean by _same environment_? i.e., same Conda or Python virtual environment

Comment (ajbozarth) [reply docs/source/developer_guide/development-workflow.md:56]: I was trying to figure out how to word that without assuming they were using conda. would saying `python environment` make sense? Personally I never had a concept of separate environments in my dev until I started working on lab and installed conda.

Comment (vabarbosa) [reply docs/source/developer_guide/development-workflow.md:56]: yes, i think `Python environment` makes sense

Comment (ajbozarth) [reply docs/source/developer_guide/development-workflow.md]: updated

**Comment by lresende** [comment]
I would prefer if we do `make install` to produce the necessary npm packages and the full wheel that is ready to install.  I am open to having something like `make dev` to do a simple build`. Otherwise, this looks good.

**Comment by ajbozarth** [comment]
Addressed @lresende comments based on discussion during scrum. We will remove creation of dists in another pr when addressing build speed and/or release automation.

**Pull request merged**